contracts/core/04-channel/IBCChannelUpgrade.sol

```diff
@@ -18,200 +19,30 @@ import {IIBCModuleUpgrade} from "../26-router/IIBCModuleUpgrade.sol";
-            )
-        ) {
-            revert IBCConnectionIBCVersionNotSupported();
-        }
-
-        // there exists at least one valid proposed change to the existing channel fields
-        if (
-            channel.ordering == proposedUpgradeFields.ordering
-                && keccak256(bytes(channel.version)) == keccak256(bytes(proposedUpgradeFields.version))
-                && keccak256(abi.encodePacked(proposedUpgradeFields.connection_hops[0]))
-                    == keccak256(abi.encodePacked(channel.connection_hops[0]))
-        ) {
-            revert IBCChannelUpgradeNoChanges();
-        }
-    }
-
-    /**
-     * @dev startFlushUpgradeHandshake will verify that the channel
-     * is in a valid precondition for calling the startFlushUpgradeHandshake.
-     * it will set the channel to flushing state.
-     * it will store the nextSequenceSend and upgrade timeout in the upgrade state.
-     */
-    function startFlushUpgradeHandshake(
-        IIBCModuleUpgrade module,
-        Channel.Data storage channel,
-        Upgrade.Data storage upgrade,
-        ChannelStorage storage channelStorage,
-        string calldata portId,
-        string calldata channelId
-    ) internal {
-        Timeout.Data memory upgradeTimeout = module.getUpgradeTimeout(portId, channelId);
-        if (
-            !(
-                upgradeTimeout.height.revision_number > 0 || upgradeTimeout.height.revision_height > 0
-                    || upgradeTimeout.timestamp > 0
-            )
-        ) {
-            revert IBCChannelUpgradeInvalidTimeout();
-        }
-        channel.state = Channel.State.STATE_FLUSHING;
-        upgrade.timeout = upgradeTimeout;
-        upgrade.next_sequence_send = channelStorage.nextSequenceSend;
-    }
-
+/**
+ * @dev IBCChannelUpgradeBase is a base contract that provides common functions for 04-channel implementations.
+ */
+abstract contract IBCChannelUpgradeBase is IBCModuleManager, IIBCChannelUpgradeBase, IIBCChannelUpgradeErrors {
     /**
      * @dev restoreChannel will restore the channel state to its pre-upgrade state
      * and delete upgrade auxiliary state so that upgrade is aborted.
      * it writes an error receipt to state so counterparty can restore as well.
      * NOTE: this function signature may be modified by implementors to take a custom error
      */
-    // reason
     function restoreChannel(string calldata portId, string calldata channelId, UpgradeHandshakeError err) internal {
         ChannelStorage storage channelStorage = getChannelStorage()[portId][channelId];
         Channel.Data storage channel = channelStorage.channel;
         channel.state = Channel.State.STATE_OPEN;
 
         delete channelStorage.upgrade;
-        deleteUpgradeCommitment(portId, channelId);
-        revertCounterpartyUpgrade(channelStorage.recvStartSequence);
+        revertCounterpartyUpgrade(channelStorage);
+        delete channelStorage.counterpartyUpgradeTimeout;
```

Comment:
This call of `delete` is redundant. The same operation is done in `revertCounterpartyUpgrade`.